### DISCUSSION TITLE
drawing-graphs-in-github-comments-with-mermaid-diagrams/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#125] [Announcements]
# drawing-graphs-in-github-comments-with-mermaid-diagrams/

In this post I show how I hacked graphs into GitHub PR comments using Mermaid diagrams

https://andrewlock.net/drawing-graphs-in-github-comments-with-mermaid-diagrams/

**Comment by satrapu** [comment]
Hey Andrew,

Nice article!
By the way, I would love to know more about how did you manage to push code coverage and benchmarks reports to GitHub via comments; could you share more details about these 2 things?

Thanks,
Bogdan

**Reply by andrewlock** [reply]
Thanks! This is all based on our build for the Datadog .NET tracer which you can find here: https://github.com/DataDog/dd-trace-dotnet/tree/master/tracer/build/_build. It's... a bit scary tbh 😅 Hoping to clean it up and blog about it at some point!

**Comment by RattanOnly** [comment]
This is such a clever hack! I've been looking for a good <a href="https://mermaidviewer.com/">mermaid diagram tutorial</a> for using it in dev workflows, and this post is perfect. The idea of using Gantt charts to make bar graphs for GitHub comments is genius. It's a bit of a workaround, but it works! Definitely going to try this out in our own CI pipeline to visualize some test results. Thanks for sharing this trick!
